From c1c593378cf96fc501b7f723727a468871db9cb1 Mon Sep 17 00:00:00 2001
From: Jayanthi A <jayanthi.annadurai@broadcom.com>
Date: Wed, 21 Nov 2012 06:53:24 +0530
Subject: [PATCH 426/565] XLP3xx: SRIO Response timeout initialization for
 XLP3xx-B1

Response timeout need to be programmed at the time of SRIO
initialization because the hardware reset value is zero.

Based on Broadcom SDK 2.3.

Signed-off-by: Jayanthi A <jayanthi.annadurai@broadcom.com>
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/mips/netlogic/xlp/xlp_srio.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/mips/netlogic/xlp/xlp_srio.c b/arch/mips/netlogic/xlp/xlp_srio.c
index 03ec5cb..d923337 100644
--- a/arch/mips/netlogic/xlp/xlp_srio.c
+++ b/arch/mips/netlogic/xlp/xlp_srio.c
@@ -1547,6 +1547,12 @@ int bcm_init_rio(void)
         else {
                 bcm_write_srio_devreg(cfgbase, SRIO_DATAMSG_SIZE, 0xE);  // 0xE - 256 Bytes is maximum segment size
                 regindex = DOORBELL_DESTID;
+                if (!is_nlm_xlp3xx_b0()) {
+                        bcm_write_srio_reg32(membase, RIO_RESPTOUT_CSR, 0xFFFFFF);
+			for(i = 0 ; i<SRIO_MAX_PORTS; i++) {
+                        	bcm_write_srio_devreg(cfgbase, FLUSHADDRQ + i, 0xFFFF);
+			}
+                }
         }
 
         bcm_write_srio_devreg(cfgbase, SRIOIPCTRL, (4<<21 )| (1 <<20) | (0xF << 16) | (0xF << 12) |  (0xF << 4));
-- 
1.8.4.93.g57e4c17

